### DISCUSSION TITLE
How to tell vitest a snapshot fails

### DISCUSSION THREAD

**Discussion by unional** [vitest-dev/vitest#6828] [Q&A] [answered]
I'm working on https://github.com/repobuddy/storybook-addon-vis.

I can get the `updateSnapshot` option from:

```ts
import { server } from '@vitest/browser/context'

server.config.snapshotOptions.updateSnapshot
```

now my problem is how to tell `vitest` that the snapshot testing fails so `vitest` will show `press u to update snapshot, press h to show help`
 
currently I am returning:

```ts
return {
  pass: false,
  message: () => '...'
}
```

**Comment by sheremet-va** [comment, marked as answer]
The `press u to update snapshot` string is displayed if there is at least one test with an error message that maches a regexp `/Snapshot .* mismatched/`

You can also probably tinker with the snapshot state via `expect.getState().stapshotState` (can't help you there)

**Reply by unional** [reply]
Awesome. It works like a charm.
